src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobiePetRepositoryInterpreter.scala

```diff
@@ -68,9 +74,14 @@ class DoobiePetRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
     extends PetRepositoryAlgebra[F] {
   import PetSQL._
 
-  def put(pet: Pet): F[Pet] =
+  def create(pet: Pet): F[Pet] =
     insert(pet).withUniqueGeneratedKeys[Long]("ID").map(id => pet.copy(id = id.some)).transact(xa)
 
+  override def update(pet: Pet): F[Option[Pet]] = {
+    def doUpdate(intoId: Long): ConnectionIO[Pet] = PetSQL.update(pet, intoId).run.map(_ => pet)
+    pet.id.traverse(doUpdate).transact(xa)
+  }
+
   def get(id: Long): F[Option[Pet]] = select(id).option.transact(xa)
 
   def delete(id: Long): F[Option[Pet]] = OptionT(get(id)).semiflatMap(pet =>
```

Comment:
I was able to change out the `semiflatMap` in the delete for the following.  Is this ok?

```
  def delete(id: Long): F[Option[Pet]] = {
    for {
      p <- get(id)
      _ <- PetSQL.delete(id).run.transact(xa)
    } yield p
  }
```

Reply:
yeah I think if a method provided by cats fits a use case we have very well, then we should not shy from using it.  The use case itself helps to instruct people of its use.